### PR TITLE
feat: support appgroup level tps

### PR DESCRIPTION
**Pull request by bentol**

### CODE REVIEW

File: flow/grpc_producer.go [hunk 1]
```diff
@@ -157,31 +159,6 @@ func (s *producerService) Start() (err error) {
 	return grpcSrv.Serve(lis)
 }
 
-func (s *producerService) LaunchREST() (err error) {
```
Comment (fadlinurhasan) [inline flow/grpc_producer.go:160]: is this function removing intended mas?

File: flow/grpc_producer.go [hunk 2]
```diff
@@ -321,3 +298,11 @@ func (s *producerService) handleProduce(timber *pb.Timber, topic string) (err er
 	prome.IncreaseKafkaMessagesStoredTotal(topic)
 	return
 }
+
+func (s *producerService) getRateLimitInfo(context *pb.TimberContext) (string, int32) {
+	if context.GetDisableAppTps() {
```
Comment (fadlinurhasan) [inline flow/grpc_producer.go:303]: where this information provided?

File: flow/grpc_producer.go [hunk 3]
```diff
@@ -24,16 +23,18 @@ const (
...
 	ErrRegisterGrpc           = errkit.Error("Error registering gRPC server endpoint into reverse proxy")
 	ErrReverseProxy           = errkit.Error("Error serving REST reverse proxy")
+
+	RateLimitKeyAppGroup = "app_group"
 )
 
 type ProducerService interface {
-	pb.ProducerServer
 	Start() error
 	LaunchREST() error
```
Comment (fadlinurhasan) [inline flow/grpc_producer.go:32]: should we also clean up this one? or this still needed?

Comment (bentol) [reply flow/grpc_producer.go]: you mean the `LaunchREST() error` right ?
updated

**Pull request merged**